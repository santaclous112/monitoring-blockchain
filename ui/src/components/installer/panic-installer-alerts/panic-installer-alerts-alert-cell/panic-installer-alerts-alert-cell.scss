.panic-installer-alerts-alert-cell {
  &__container {
    display: flex;
    flex-direction: column;

    svc-label {
      font-weight: 900;
      margin-left: 3px;
      margin-right: 5px;

      ion-label {
        text-transform: capitalize;
      }
    }

    svc-text {
      font-size: 0.8em !important;
      margin-left: 10px;
    }
  }

  &__enabled-input {
    display: flex;
    flex-direction: row;
    align-items: flex-start;

    svc-checkbox {
      ion-item {
        height: 0px;
        --min-height: 0px;
      }

      ion-checkbox {
        margin: 0px !important;
        position: fixed;
      }
    }
  }
}

ion-item {
  margin-left: -12px;
  text-transform: capitalize;
}
