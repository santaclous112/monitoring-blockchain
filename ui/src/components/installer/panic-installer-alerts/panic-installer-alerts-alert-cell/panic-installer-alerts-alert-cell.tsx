import {Component, h, Prop} from '@stencil/core';
import {SeverityType} from "../../../../../../entities/ts/SeverityType";

@Component({
  tag: 'panic-installer-alerts-alert-cell',
  styleUrl: 'panic-installer-alerts-alert-cell.scss'
})
export class PanicInstallerAlertsAlertCell {

  /**
   * the alert identifier
   */
  @Prop() alertIdentifier: string;
  /**
   * the SeverityType object associated with the alert
   */
  @Prop() severity: SeverityType;
  /**
   * the default state of the severity enabled setting
   */
  @Prop() severityEnabled: boolean;
  /**
   * the method in which time is measured, for example: seconds; blocks; eras; etc.
   */
  @Prop() adornment: string;
  /**
   * the title of the primary input
   */
  @Prop() primaryInputLabel: string;
  /**
   * the default text of the primary input
   */
  @Prop() primaryInputValue: number;
  /**
   * the title of the secondary input
   */
  @Prop() secondaryInputLabel: string;
  /**
   * the default text of the secondary input
   */
  @Prop() secondaryInputValue: number;
  /**
   * the default text of the tertiary input
   */
  @Prop() tertiaryInputLabel: string;
  /**
   * the default text of the tertiary input
   */
  @Prop() tertiaryInputValue: number;
  /**
   * the default state of the repeat enabled setting
   */
  @Prop() repeatEnabled: boolean;

  /**
   * Renders a form which is dynamically built depending on the props that have been passed,
   * including the set of default values for each prop. Each alert setting is classified by
   * a combination of 3 lowercase attributes: identifier, severity and field. The combination
   * of these attributes allows the component to uniquely identify each alert as well as each of
   * its specific properties being adjusted, allowing it to be then updated in the config.
   */
  render(){
    return (
      <div class={"panic-installer-alerts-alert-cell__container"}>
        <div class={"panic-installer-alerts-alert-cell__enabled-input"}>
          <svc-label>{`${this.severity.name} Enabled`}</svc-label>
          <panic-installer-form-checkbox
              name={`${this.alertIdentifier} ${this.severity.value} enabled`}
              checked={this.severityEnabled}
          />
        </div>

        <svc-input
          name={`${this.alertIdentifier} ${this.severity.value} ${this.primaryInputLabel}`}
          type={"number"}
          labelPosition={"floating"}
          label={`${this.primaryInputLabel} (${this.adornment})`}
          value={this.primaryInputValue}
        />

        {
          this.secondaryInputLabel && this.secondaryInputLabel.includes("repeat") &&
            <div class={"panic-installer-alerts-alert-cell__enabled-input"}>
              <svc-label>Repeat Enabled</svc-label>
              <panic-installer-form-checkbox
                  name={`${this.alertIdentifier} ${this.severity.value} repeatEnabled`}
                  checked={this.repeatEnabled}
              />
            </div>
        }

        {
          this.secondaryInputLabel &&
            <svc-input
              name={`${this.alertIdentifier} ${this.severity.value} ${this.secondaryInputLabel}`}
              type={"number"}
              labelPosition={"floating"}
              label={`${this.secondaryInputLabel} (${this.adornment})`}
              value={this.secondaryInputValue}
            />
        }

        {
          this.tertiaryInputLabel &&
            <svc-input
              name={`${this.alertIdentifier} ${this.severity.value} ${this.tertiaryInputLabel}`}
              type={"number"}
              labelPosition={"floating"}
              label={`${this.tertiaryInputLabel} (${this.adornment})`}
              value={this.tertiaryInputValue}
            />
        }
      </div>
    )
  }
}
