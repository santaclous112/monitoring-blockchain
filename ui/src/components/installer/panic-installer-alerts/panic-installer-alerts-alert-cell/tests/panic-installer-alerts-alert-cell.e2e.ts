import { newSpecPage } from '@stencil/core/testing';
import {PanicInstallerAlertsAlertCell} from "../panic-installer-alerts-alert-cell";

describe('panic-installer-alerts-alert-cell', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [PanicInstallerAlertsAlertCell],
      html: '<panic-installer-alerts-alert-cell alertIdentifier="test" severity="Warning" severityEnabled="true" primaryInputLabel="test" primaryInputValue="100"></panic-installer-alerts-alert-cell>',
    });

    const panicInstallerAlertsAlertCell = page.body.querySelector('panic-installer-alerts-alert-cell');
    expect(panicInstallerAlertsAlertCell).toMatchSnapshot();
  });
});