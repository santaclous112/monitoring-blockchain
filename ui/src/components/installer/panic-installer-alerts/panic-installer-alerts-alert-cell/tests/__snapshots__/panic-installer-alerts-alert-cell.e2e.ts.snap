// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`panic-installer-alerts-alert-cell renders 1`] = `
<panic-installer-alerts-alert-cell alertidentifier="test" primaryinputlabel="test" primaryinputvalue="100" severity="Warning" severityenabled="true">
  <div class="panic-installer-alerts-alert-cell__container">
    <div class="panic-installer-alerts-alert-cell__enabled-input">
      <svc-label>
        undefined Enabled
      </svc-label>
      <panic-installer-form-checkbox name="undefined undefined enabled"></panic-installer-form-checkbox>
    </div>
    <svc-input label="undefined (undefined)" labelposition="floating" name="undefined undefined undefined" type="number"></svc-input>
  </div>
</panic-installer-alerts-alert-cell>
`;
